pkg/cache/cache.go

```diff
@@ -855,19 +856,22 @@ func (c *Cache) ClusterQueuesUsingFlavor(flavor kueue.ResourceFlavorReference) [
 	return cqs
 }
 
-func (c *Cache) ClusterQueuesUsingTopology(tName kueue.TopologyReference) []kueue.ClusterQueueReference {
-	c.RLock()
-	defer c.RUnlock()
-	var cqs []kueue.ClusterQueueReference
+func (c *Cache) ClusterQueuesUsingTopology(tName kueue.TopologyReference) iter.Seq[kueue.ClusterQueueReference] {
+	return func(yield func(kueue.ClusterQueueReference) bool) {
+		c.RLock()
+		defer c.RUnlock()
 
-	for _, cq := range c.hm.ClusterQueues() {
-		for _, tRef := range cq.tasFlavors {
-			if tRef == tName {
-				cqs = append(cqs, cq.Name)
+		for _, cq := range c.hm.ClusterQueues() {
+			for _, tRef := range cq.tasFlavors {
+				if tRef != tName {
+					continue
+				}
+				if !yield(cq.Name) {
+					return
+				}
```

Comment:
I synced with @mimowo a bit about this change.
This will increase the locking time a bit since this iterator will hold until the caller finishes all iterations throughout range-over-func, we guess.

It is probably not significant problem, but we decided to drop this change from this PR, and open a follow-up PR so that we can appropriately evaluate this change.

Reply:
Done.